### PR TITLE
[JUJU-1444] Update go.mod with semantic versions.

### PR DESCRIPTION
**Pull request by juanmanuel-tirado**
Refactor `go.mod` to use released versions instead of commit hashes. This change applies to most of the consumed repos. Some of them such as blobstore, lumberjack or mgo remain for future PRs.

## Checklist

NA

## QA steps

Trust the correctness of the CI.

## Documentation changes

NA

## Bug reference

NA

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -49,48 +49,48 @@ require (
...
-	github.com/juju/idmclient/v2 v2.0.0-20220207024613-525e1ac3a890
-	github.com/juju/jsonschema v0.0.0-20220207021905-6f321f9146b3
-	github.com/juju/jsonschema-gen v0.0.0-20220207021829-b1697c753155
-	github.com/juju/loggo v0.0.0-20210728185423-eebad3a902c4
+	github.com/juju/http/v2 v2.0.0
+	github.com/juju/idmclient/v2 v2.0.0
+	github.com/juju/jsonschema v1.0.0
+	github.com/juju/jsonschema-gen v1.0.0
+	github.com/juju/loggo v1.0.0
 	github.com/juju/lumberjack v2.0.0-20200420012306-ddfd864a6ade+incompatible
```
Comment (wallyworld) [inline go.mod:70]: lumberjack can use v2.0.1 right? Which corresponds to commit ddfd864a6ade

File: go.mod [hunk 2]
```diff
@@ -49,48 +49,48 @@ require (
...
-	github.com/juju/gojsonschema v0.0.0-20150312170016-e1ad140384f2
+	github.com/juju/clock v1.0.0
+	github.com/juju/cmd/v3 v3.0.0
+	github.com/juju/collections v1.0.0
+	github.com/juju/description/v3 v3.0.1
+	github.com/juju/errors v1.0.0
+	github.com/juju/featureflag v1.0.0
+	github.com/juju/gnuflag v1.0.0
+	github.com/juju/gojsonschema v1.0.0
 	github.com/juju/gomaasapi/v2 v2.0.0-20220215192228-f1f389204471
```
Comment (wallyworld) [inline go.mod:64]: gomaasapi is a few commits behind tip of v2 branch
and there's no tags - could we add a tag to the gomaasapi branch and then use that here

File: go.mod [hunk 3]
```diff
@@ -49,48 +49,48 @@ require (
 	github.com/hashicorp/raft v1.3.2-0.20210825230038-1a621031eb2b
 	github.com/hashicorp/raft-boltdb v0.0.0-20171010151810-6e5ba93211ea
 	github.com/im7mortal/kmutex v1.0.1
-	github.com/juju/ansiterm v0.0.0-20210929141451-8b71cc96ebdc
+	github.com/juju/ansiterm v1.0.0
 	github.com/juju/blobstore/v2 v2.0.0-20220204135513-4876189e56a4
 	github.com/juju/charm/v8 v8.0.0-20220509231111-ed6d505a46f4
```
Comment (wallyworld) [inline go.mod:54]: charm/v8 has a v8.0.0 tag which points to an older commit
could we add a new v8.0.1 tag and use that here

**Review by wallyworld — COMMENTED** [review]
Looks good - I checked a few of the commit hashes and they seemed to line up with the tags used.
Can we complete the job by looking at the remaining upstream juju branches and either use the tags that are there or add the missing tags - I've given a couple of examples in the comments

File: go.mod (hunk 1 above)
Comment (wallyworld) [reply go.mod:70]: Maybe we can add a go.mod upstream and lose the "incompatible"

Comment (juanmanuel-tirado) [reply go.mod:70]: Added a go.mod and tagged a v2.0.2.

File: go.mod (hunk 2 above)
Comment (juanmanuel-tirado) [reply go.mod:64]: I released a v2.0.0 with the corresponding tag.

**Review by wallyworld — APPROVED** [review]

**Pull request merged**